common/txmgr/types/orm.go

```diff
@@ -0,0 +1,95 @@
+	// FindEthTxWithAttempts(etxID TXID, opts ...any) (etx TX, err error)
+	// FindEthTxWithNonce(fromAddress ADDR, nonce TXMETA, opts ...any) (etx *TX, err error)
+	// FindNextUnstartedTransactionFromAddress(etx *TX, fromAddress ADDR, chainID CHAINID, opts ...any) error
+	// FindTransactionsConfirmedInBlockRange(highBlockNumber, lowBlockNumber int64, chainID CHAINID, opts ...any) (etxs []*TX, err error)
+	// GetEthTxInProgress(fromAddress ADDR, opts ...any) (etx *TX, err error)
+	// GetInProgressEthTxAttempts(ctx context.Context, address ADDR, chainID CHAINID, opts ...any) (attempts []TXATTEMPT, err error)
+	// HasInProgressTransaction(account ADDR, chainID CHAINID, opts ...any) (exists bool, err error)
+	// // InsertEthReceipt only used in tests. Use SaveFetchedReceipts instead
+	InsertEthReceipt(receipt *Receipt[R, HASH], opts ...any) error
+	// InsertEthTx(etx *TX, opts ...any) error
+	// InsertEthTxAttempt(attempt *TXATTEMPT, opts ...any) error
+	// LoadEthTxAttempts(etx *TX, opts ...any) error
+	// LoadEthTxesAttempts(etxs []*TX, opts ...any) error
+	// MarkAllConfirmedMissingReceipt(chainID CHAINID, opts ...any) (err error)
+	// MarkOldTxesMissingReceiptAsErrored(blockNum int64, finalityDepth uint32, chainID CHAINID, opts ...any) error
+	// PreloadEthTxes(attempts []TXATTEMPT, opts ...any) error
+	PruneUnstartedEthTxQueue(queueSize uint32, subject uuid.UUID, opts ...any) (n int64, err error)
+	// SaveConfirmedMissingReceiptAttempt(ctx context.Context, timeout time.Duration, attempt *TXATTEMPT, broadcastAt time.Time, opts ...any) error
+	SaveFetchedReceipts(receipts []R, chainID CHAINID, opts ...any) (err error)
+	// SaveInProgressAttempt(attempt *TXATTEMPT, opts ...any) error
+	// SaveInsufficientEthAttempt(timeout time.Duration, attempt *TXATTEMPT, broadcastAt time.Time, opts ...any) error
+	// SaveReplacementInProgressAttempt(oldAttempt TXATTEMPT, replacementAttempt *TXATTEMPT, opts ...any) error
+	// SaveSentAttempt(timeout time.Duration, attempt *TXATTEMPT, broadcastAt time.Time, opts ...any) error
+	// SetBroadcastBeforeBlockNum(blockNum int64, chainID CHAINID, opts ...any) error
+	// UpdateBroadcastAts(now time.Time, etxIDs []TXID, opts ...any) error
+	// UpdateEthKeyNextNonce(newNextNonce, currentNextNonce TXMETA, address ADDR, chainID CHAINID, opts ...any) error
+	// UpdateEthTxAttemptInProgressToBroadcast(etx *TX, attempt TXATTEMPT, NewAttemptState TxAttemptState, incrNextNonceCallback CallbackFunc, opts ...any) error
+	// UpdateEthTxsUnconfirmed(ids []int64, opts ...any) error
+	// UpdateEthTxUnstartedToInProgress(etx *TX, attempt *TXATTEMPT, opts ...any) error
+	// UpdateEthTxFatalError(etx *TX, opts ...any) error
+	// UpdateEthTxForRebroadcast(etx TX, etxAttempt TXATTEMPT, opts ...any) error
+	// Close()
+}
+
+// TxStrategy controls how txes are queued and sent
+//
+//go:generate mockery --quiet --name TxStrategy --output ./mocks/ --case=underscore --structname TxStrategy --filename tx_strategy.go
+type TxStrategy interface {
```

Comment:
This was previously not chain/db-agnostic so I brought it here.

I genearlized this because I originally wanted to generalize `txmgr.NewTx`  in this PR. (txmgr.NewTx` is *dependent* on `TxStrategy`) but it's out of scope for PR. However, `TxStrategy` needs to be generalized anyway and I already did the work to do it.